apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -387,16 +394,16 @@ public override Task StartAsync(Func<PubsubMessage, CancellationToken, Task<Repl
             Action<Task> registerTask = task =>
             {
                 registeredTasks.Locked(() => registeredTasks.Add(task));
-                Action unregisterTask = () => registeredTasks.Locked(() => registeredTasks.Remove(task));
-                _taskHelper.Run(async () => await _taskHelper.ConfigureAwaitWithFinally(() => task, unregisterTask));
+                Action<Task> unregisterTask = t => registeredTasks.Locked(() => registeredTasks.Remove(t));
+                task.ContinueWith(unregisterTask, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, _taskHelper.TaskScheduler);
```

Comment:
Does it make sense to use ExecuteSynchronously and specify a task scheduler? Isn't the point of ExecuteSynchronously that it just keeps going on the same thread, with no scheduling needed? Maybe I've misunderstood.